generator/uscore/metadata_extractor.rb

```diff
@@ -206,11 +206,30 @@ def add_search_param_descriptions(profile_definition, sequence)
           profile_element = profile_definition['snapshot']['element'].select { |el| el['id'] == path }.first
           param_metadata = {
             path: path,
-            comparators: {}
+            comparators: {},
+            values: []
           }
           if !profile_element.nil?
             param_metadata[:type] = profile_element['type'].first['code']
             param_metadata[:contains_multiple] = (profile_element['max'] == '*')
+            if param_metadata[:contains_multiple]
```

Comment:
I moved it to another function. It checks a lot of different stuff so not sure how to make simpler than it currently is.

Reply:
Wherever there is a comment explaining what the next few lines do, it could instead be a method whose name says what it does.
```
if param_metadata[:contains_multiple]
  add_values_from_slices(args)
elsif param_metadata[:type] == 'CodeableConcept'
  add_values_from_codeable_concepts(args)
end
add_whatever_this_one_is(args)
if value_set_binding
  add_values_from_valueset(args)
end
add_values_from_metadata unless ...
```

Reply:
Fixed